qt/widgets/common/src/MuonFitPropertyBrowser.cpp

```diff
@@ -1483,12 +1499,110 @@ void MuonFitPropertyBrowser::clearChosenPeriods() const {
     m_boolManager->setValue(iter.value(), false);
   }
 }
+/**
+* updates the period displays
+*/
+void MuonFitPropertyBrowser::updatePeriods() {
+  int j = m_enumManager->value(m_periodsToFit);
+  // auto selected = getChosenPeriods();
+  updatePeriods(j);
+}
+/**
+* updates the period displays and conserves the selection
+* if selection is niot available default to all periods
+* @param j :: [input] index of selection in combobox
+* selected is an input for changing runs and preserving selection (list of
+* selected periods)
+* currently has a bug
+*/
+void MuonFitPropertyBrowser::updatePeriods(const int j) {
+  // this is for switching but has a bug at the moment
+  // const QStringList &selected) {
+  if (m_periodsToFitOptions[j] == CUSTOM_LABEL) {
+    // currently the below does not work reliably (if period arithmatic is
+    // presemnt it gives bad results
+    /*
+    m_enumManager->setEnumNames(m_periodsToFit, m_periodsToFitOptions);
+    m_enumManager->setValue(m_periodsToFit, j);
+    setChosenPeriods(selected);*/
+    // lets default to all periods for now
+    m_enumManager->setEnumNames(m_periodsToFit, m_periodsToFitOptions);
+    m_enumManager->setValue(m_periodsToFit, j);
+    // explictly set all periods
+    setAllPeriods();
+  } else if (m_periodsToFitOptions[j] == ALL_PERIODS_LABEL) {
+    m_enumManager->setEnumNames(m_periodsToFit, m_periodsToFitOptions);
+    m_enumManager->setValue(m_periodsToFit, j);
+    // explictly set all periods
+    setAllPeriods();
+  } else { // single number
+    m_enumManager->setEnumNames(m_periodsToFit, m_periodsToFitOptions);
+    m_enumManager->setValue(m_periodsToFit, j);
```

Comment:
It looks like these two lines are always executed regardless of the if/else statements so shouldn't they be moved outside instead of duplicated?